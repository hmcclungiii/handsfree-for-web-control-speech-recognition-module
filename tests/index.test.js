import module from '../src/index'

function mockExtension(methods = {}) {
  global.chrome = {
    extension: {
      sendMessage: jest.fn(),
      ...methods
    }
  }
}

describe('Module', () => {
  it('defines a module', () => {
    expect(module).toMatchSnapshot()
  })
  describe('startListening action', () => {
    it('calls background action', () => {
      const sendMessage = jest.fn()
      mockExtension({ sendMessage })
      const startListening = module.contexts[0].commands[0]
      expect(startListening.name).toEqual('i18n-command.start-listening')
      startListening.action()
      expect(sendMessage.mock.calls).toMatchSnapshot()
    })
  })
  describe('stopListening action', () => {
    it('calls background action', () => {
      const sendMessage = jest.fn()
      mockExtension({ sendMessage })
      const stopListening = module.contexts[0].commands[1]
      expect(stopListening.name).toEqual('i18n-command.stop-listening')
      stopListening.action()
      expect(sendMessage.mock.calls).toMatchSnapshot()
    })
  })
})
