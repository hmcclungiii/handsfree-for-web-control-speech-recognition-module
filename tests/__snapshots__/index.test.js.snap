// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module defines a module 1`] = `
Object {
  "contexts": Array [
    Object {
      "commands": Array [
        Object {
          "action": [Function],
          "group": "i18n-group.listeningControl",
          "help": "i18n-help.start-listening",
          "name": "i18n-command.start-listening",
        },
        Object {
          "action": [Function],
          "group": "i18n-group.listeningControl",
          "help": "i18n-help.stop-listening",
          "name": "i18n-command.stop-listening",
        },
      ],
      "context": "root",
      "i18n": Object {
        "en": Object {
          "command.start-listening": "start listening",
          "command.stop-listening": "stop listening",
          "group.listeningControl": "Listening Control",
          "help.start-listening": "Start voice recognition process",
          "help.stop-listening": "Stop voice recognition process",
        },
        "es": Object {
          "command.start-listening": "iniciar reconocimiento",
          "command.stop-listening": "detener reconocimiento",
          "group.listeningControl": "Control de reconocimiento de voz",
          "help.start-listening": "Inicia el proceso de reconocimiento de voz",
          "help.stop-listening": "Concluye el proceso de reconocimiento de voz",
        },
      },
    },
  ],
  "description": "i18n-description",
  "i18n": Object {
    "en": Object {
      "description": "Commands useful to tell to Handsfree for web when to listen for voice commands.",
      "name": "Speech Recognition Control",
    },
    "es": Object {
      "description": "Comandos útiles para indicarle a la extensión cuando escuchar comandos de voz.",
      "name": "Control de reconocimiento de voz",
    },
  },
  "icon": "fa fa-search-plus",
  "name": "i18n-name",
}
`;

exports[`Module startListening action calls background action 1`] = `
Array [
  Array [
    Object {
      "startSpeechRecognition": true,
    },
    [Function],
  ],
]
`;

exports[`Module stopListening action calls background action 1`] = `
Array [
  Array [
    Object {
      "startSleepMode": true,
    },
    [Function],
  ],
]
`;
